CONTRIBUTING.md

```diff
@@ -42,8 +44,90 @@ mvn -f pom.client.xml -Dgpg.skip -DskipTests clean install
 
 //example: mvn -f sdk/keyvault/azure-keyvault-keys/pom.xml clean install
 ```
+
+## Versions and versioning
+
+Tooling has been introduced to centralize versioning and help ease the pain of updating artifact versions in POM and README files. Under the eng\versioning directory there version text files, one for client (version_client.txt) and one for data (version_data.txt). The format of the version files is as follows:
+groupId:artifactId;dependency-version;current-version
+The dependency-version should be set to the most recent released version and the current-version is set to the next version to be released.
+
+For example:
+
+`com.azure:azure-identity;1.0.0-preview.4;1.0.0-preview.5`
+
+Note: In the case of a new or unreleased artifact both versions will be the same.
+
+For more information 
+
+### Libraries vs External Dependencies
+
+Libraries refer to things that are built and released as part of the Azure SDK. Libraries have a current version and a dependency version. 
+
+External Dependencies refer to dependencies for things that are not built and released as part of the Azure SDK regardless of the source. External Dependencies will only ever have one version.
+
+### Current version vs Depdendency version
+
+Current version - This is the version we should be using when defining a component in its POM file and also when dependent components are built within the same pipeline. The current version is the version currently in development.
+Depdendency version - This is the version we should be using when a given library is a dependency outside of a particular area. This should be the latest released version of the package.
+
+For example: `com.azure:azure-storage-blob-batch` has dependencies on `com.azure:azure-core`, `com.azure:azure-core-http-netty` and `com.azure:azure-storage-blob`. Because `com.azure:azure-core` and `com.azure:azure-core-http-netty` are both built outside of azure-storage pipeline we should be using the released or *Dependency* versions of these when they're dependencies of another library. Similarly, libraries built as part of the same pipeline, that have interdependencies, should be using the Current version. Since `com.azure:azure-storage-blob-batch` and `com.azure:azure-storage-blob` are both built part of the azure-batch pipeline when `com.azure:azure-storage-blob` is declared as a dependency of `com.azure:azure-storage-blob-batch` it should be the *Current* version.
+
+This is going to be especially important after GA when releases aren't going to be the entire Azure SDK every time. If we're releasing a patch for a targeted azure-storage fix then we shouldn't need to build and release azure-core, we should be targeting the released versions and only building/releasing that update to azure-storage. It's worth noting that right now, in the version_client.txt, the dependency/current versions are the same. This will change once we GA, at which point the current version should be ahead of the dependency version.
```

Comment:
You will want to document how developers who are fixing an issue in azure-core can then consume this fix on their machine such that they can use it in azure-storage-blob, for example. If they are only depending on released libraries, they need to have a way to circumvent that so that they may continue to work with what exists in the repo (or on their machines before they even push to the repo).

Reply:
Updated in the next iteration of https://github.com/Azure/azure-sdk-for-java/pull/5889